### PR TITLE
[lifecycleExperimental] Issue #1303

### PR DESCRIPTION
**Pull request by maurocarrero**
Removes lifecycleExperimental from ShallowWrapper docs.
~~Updates code throwing an error when this option is used.~~
~~Updates related unit tests.~~
Updates CONTRIBUTING doc.

### CODE REVIEW

File: docs/api/shallow.md [hunk 1]
```diff
@@ -46,10 +46,7 @@ describe('<MyComponent />', () => {
   - `options.context`: (`Object` [optional]): Context to be passed into the component
   - `options.disableLifecycleMethods`: (`Boolean` [optional]): If set to true, `componentDidMount`
 is not called on the component, and `componentDidUpdate` is not called after
-[`setProps`](ShallowWrapper/setProps.md) and [`setContext`](ShallowWrapper/setContext.md). Default to `false`. 
-  - `options.lifecycleExperimental`: (`Boolean` [optional]): If set to true, the entire lifecycle
-(`componentDidMount` and `componentDidUpdate`) of the React component is called. The current default value
-is `false` with enzyme v2, but the next major version will flip the default value to `true`.
+[`setProps`](ShallowWrapper/setProps.md) and [`setContext`](ShallowWrapper/setContext.md). Default to `false`.
```
Comment (maurocarrero) [inline docs/api/shallow.md]: I removed lifecycleExperimental mentions but perhaps you want to add some info related to the fat that now shallow has this behaviour by default.

File: packages/enzyme-test-suite/test/ShallowWrapper-spec.jsx [hunk 2]
```diff
@@ -3117,74 +3108,15 @@ describe('shallow', () => {
...
-        [true, false, undefined].forEach((value) => {
-          expect(() => shallow(<div />, {
-            lifecycleExperimental: value,
-          })).not.to.throw();
+          })).to.throw(/not longer supported/);
         });
       });
     });
 
-    context('mounting phase', () => {
```
Comment (maurocarrero) [inline packages/enzyme-test-suite/test/ShallowWrapper-spec.jsx]: I guess this is not needed anymore.

File: packages/enzyme-test-suite/test/ShallowWrapper-spec.jsx [hunk 3]
```diff
@@ -3527,6 +3459,13 @@ describe('shallow', () => {
             { foo: 'bar' }, { foo: 'baz' },
             { foo: 'context' },
           ]);
+        } else {
+          expected.push([
+            'componentDidUpdate',
+            { foo: 'props' }, { foo: 'props' },
+            { foo: 'bar' }, { foo: 'baz' },
+            undefined,
```
Comment (maurocarrero) [inline packages/enzyme-test-suite/test/ShallowWrapper-spec.jsx]: I do not have enough insight into enzyme to evaluate whether this is acceptable, it smells bad. I wanted to create this PR anyway so you can guide me on how this should behave and fix it (or not), the code provoking this is in ShallowWrapper.js:413. If this is as weird to you as to me, I can give it more time and research how the adapter works, that option is undefined for React16.

File: packages/enzyme/src/ShallowWrapper.js [hunk 4]
```diff
@@ -402,7 +391,7 @@ class ShallowWrapper {
         instance.setState(state, callback);
         if (
           shouldRender &&
-          this[OPTIONS].lifecycleExperimental &&
+          !this[OPTIONS].disableLifecycleMethods &&
```
Comment (maurocarrero) [inline packages/enzyme/src/ShallowWrapper.js]: This change could be the cause for the undefined in the context, my comment above. But it makes sense at first sight.

Comment (koba04) [reply packages/enzyme/src/ShallowWrapper.js]: This will be fixed at #1261

File: packages/enzyme-test-suite/test/ShallowWrapper-spec.jsx (hunk 3 above)
Comment (koba04) [reply packages/enzyme-test-suite/test/ShallowWrapper-spec.jsx]: This will be fixed at #1261

Comment (maurocarrero) [reply packages/enzyme-test-suite/test/ShallowWrapper-spec.jsx]: Will revert this change then.

**Comment by ljharb** [comment]
I'm actually not sure about the code change; it might make this a semver-major.

Perhaps it'd be better to remove it from the docs, but leave it in the API?

**Comment by maurocarrero** [comment]
Agree, it'd be a breaking change, updating only the docs now. 
As it is now those who try to use it (because they already knew the flag), could receive misleading errors and no-op behaviour.
I'd be glad to help further.

**Pull request merged**